SMLHelper/Patchers/LanguagePatcher.cs

```diff
@@ -100,84 +101,43 @@ private static void ReadOverrideCustomLines()
-                        continue; // Not correctly formatter
-                    }
-
-                    string key = split[0];
-
-                    if (!originalLines.ContainsKey(key))
-                    {
-                        Logger.Log($"Key '{key}' on line '{lineIndex}' in language override file for '{modName}' did not match an original key.", LogLevel.Warn);
-                        continue; // Skip keys we don't recognize.
-                    }
-
-                    customLines[key] = TrimTextDelimiters(split[1]);
-                    overridesApplied++;
-                }
+                int overridesApplied = ExtractCustomLinesFromText(modName, text, originalLines);
 
                 Logger.Log($"Applied {overridesApplied} language overrides to mod {modName}.", LogLevel.Info);
             }
         }
 
-        private static bool FileNeedsRewrite(string modKey)
+        internal static int ExtractCustomLinesFromText(string modName, string text, Dictionary<string, string> originalLines)
         {
-            Dictionary<string, string> modCustomLines = originalCustomLines[modKey];
-            string fileName = $"{LanguageOrigDir}/{modKey}.txt";
-
-            if (!File.Exists(fileName))
-                return true; // File not found
-
-            string[] lines = File.ReadAllLines(fileName, Encoding.UTF8);
-
-            if (lines.Length != modCustomLines.Count)
-                return true; // Difference in line count
+            MatchCollection matches = OverrideRegex.Matches(text);
 
-            // Confirm if the file actually needs to be updated
-            foreach (string line in lines)
+            int overridesApplied = 0;
+            foreach (Match match in matches)
             {
-                string[] split = line.Split(new[] { KeyValueSeparator }, StringSplitOptions.RemoveEmptyEntries);
+                string key = match.Groups["key"].Value;
+                string value = match.Groups["value"].Value;
 
-                if (split.Length != 2)
-                    return true; // Malformatted, likely externally edited
-
-                string lineKey = split[0];
-                string lineValue = TrimTextDelimiters(split[1]);
-
-                if (modCustomLines.TryGetValue(lineKey, out string origValue))
+                if (!originalLines.ContainsKey(key))
                 {
-                    if (origValue != lineValue)
-                    {
-                        return true; // Difference in line content
-                    }
-                }
-                else
-                {
-                    return true; // Key not found
+                    Logger.Log($"Key '{key}' in language override file for '{modName}' did not match an original key.", LogLevel.Warn);
+                    continue; // Skip keys we don't recognize
                 }
+
+                customLines[key] = value;
```

Comment:
Before doing this, first check if `key` begins with the prefix "Tooltip_".  
If it does, it will be necessary to override the value for this key in the doctionary `TooltipFactory.techTypeTooltipStrings.valueToString`.